src/Eloquent/Builder.php

```diff
@@ -306,19 +315,37 @@ public function createOrFirst(array $attributes = [], array $values = [])
         }
     }
 
+    /**
+     * Add subsequent queries to include an aggregate value for a relationship.
+     * For embedded relations, a projection is used to calculate the aggregate.
+     *
+     * @see \Illuminate\Database\Eloquent\Concerns\QueriesRelationships::withAggregate()
+     *
+     * @param  mixed  $relations Name of the relationship or an array of relationships to closure for constraint
+     * @param  string $column    Name of the field to aggregate
+     * @param  string $function  Required aggregation function name (count, min, max, avg)
+     *
+     * @return $this
+     */
     public function withAggregate($relations, $column, $function = null)
     {
         if (empty($relations)) {
             return $this;
         }
 
+        assert(is_string($function), new TypeError('Argument 3 ($function) passed to withAggregate must be of the type string, ' . get_debug_type($function) . ' given'));
+
         $relations = is_array($relations) ? $relations : [$relations];
 
         foreach ($this->parseWithRelations($relations) as $name => $constraints) {
             $segments = explode(' ', $name);
 
+            $alias = match (true) {
+                count($segments) === 1 => Str::snake($segments[0]) . '_' . $function,
+                count($segments) === 3 && Str::lower($segments[1]) => $segments[2],
```

Comment:
Is this missing `Str::lower($segments[1]) === 'as'`?

Reply:
Thanks for pointing, I restored the condition. And I added tests to the alias feature.